iceoryx_utils/test/moduletests/test_unit_duration.cpp

```diff
@@ -18,218 +18,1919 @@
+    auto duration = 1_s + 12_ns;
+
+    multiply(duration, 0.0, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondWithSignedResultsInLessThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{36_ns};
+    auto duration = 12_ns;
+
+    multiply(duration, 3, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondWithUnsignedResultsInLessThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{36_ns};
+    auto duration = 12_ns;
+
+    multiply(duration, 3U, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondWithFloatResultsInLessThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{42_ns};
+    auto duration = 12_ns;
+
+    multiply(duration, 3.5, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondWithSignedResultsInMoreThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{1_s + 800_ms};
+    auto duration = 600_ms;
+
+    multiply(duration, 3, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondWithUnsignedResultsInMoreThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{1_s + 800_ms};
+    auto duration = 600_ms;
+
+    multiply(duration, 3U, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondWithFloatResultsInMoreThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{2_s + 100_ms};
+    auto duration = 600_ms;
+
+    multiply(duration, 3.5, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationMoreThanOneSecondWithSignedResultsInMoreThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{13_s + 800_ms};
+    auto duration = 4_s + 600_ms;
+
+    multiply(duration, 3, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationMoreThanOneSecondWithUnsignedResultsInMoreThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{13_s + 800_ms};
+    auto duration = 4_s + 600_ms;
+
+    multiply(duration, 3U, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationMoreThanOneSecondWithFloatResultsInMoreThanOneSecond)
+{
+    constexpr Duration EXPECTED_DURATION{16_s + 100_ms};
+    auto duration = 4_s + 600_ms;
+
+    multiply(duration, 3.5, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationWithFractionalFloat)
+{
+    constexpr Duration EXPECTED_DURATION{2_s + 800_ms};
+    auto duration = 5_s + 600_ms;
+
+    multiply(duration, 0.5, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationWithNegativMultiplicatorResultsInZero)
+{
+    constexpr Duration EXPECTED_DURATION{0_s};
+    auto duration = 4_s + 600_ms;
+
+    multiply(duration, -1, EXPECTED_DURATION);
+    multiply(duration, -1.0, EXPECTED_DURATION);
+}
+
+TEST(Duration_test, MultiplyDurationLessThanOneSecondResultsInMoreNanosecondsThan64BitCanRepresent)
+{
+    constexpr uint64_t MULTIPLICATOR{(1ULL << 32U) * 42U + 73U};
+    constexpr Duration DURATION = 473_ms + 578_us + 511_ns;
+    auto EXPECTED_RESULT = DurationFactory{85428177141U, 573034055U};
```

Comment:
Hard to follow without calculator. Ideally would be computed (constexpr) from the multiplicator and the individual parts (nanoseconds and seconds) of Duration.

The point is, the nanoseconds overflow. :-)

Reply:
yes, the problem is the nanoseconds overflow. It's easier to calculate it with a calculator which supports big numbers :)